bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java

```diff
@@ -0,0 +1,198 @@
+  private static final String DATAFLOW_REGION = "region";
+
+  // Column family name used in all test bigtables.
+  private static final String CF = "cf";
+
+  // Full path of the Cloud Storage folder where dataflow jars are uploaded to.
+  private static final String GOOGLE_DATAFLOW_STAGING_LOCATION = "google.dataflow.stagingLocation";
+
+  private Connection connection;
+  private String projectId;
+  private String instanceId;
+  private String tableId;
+  private String region;
+
+  private GcsUtil gcsUtil;
+  private String dataflowStagingLocation;
+  private String workDir;
+  private byte[][] keySplits;
+
+  // Snapshot data setup
+  private String hbaseSnapshotDir;
+  private String restoreDir;
+
+  @Before
+  public void setup() throws Exception {
+    projectId = getTestProperty(BigtableOptionsFactory.PROJECT_ID_KEY);
+    instanceId = getTestProperty(BigtableOptionsFactory.INSTANCE_ID_KEY);
+    dataflowStagingLocation = getTestProperty(GOOGLE_DATAFLOW_STAGING_LOCATION);
+    region = getTestProperty(DATAFLOW_REGION);
+    String cloudTestDataFolder = getTestProperty(CLOUD_TEST_DATA_FOLDER);
+    if (!cloudTestDataFolder.endsWith(File.separator)) {
+      cloudTestDataFolder = cloudTestDataFolder + File.separator;
+    }
+
+    hbaseSnapshotDir = cloudTestDataFolder + "data/";
+    UUID test_uuid = UUID.randomUUID();
+    restoreDir = cloudTestDataFolder + "restore/" + test_uuid;
+
+    // Cloud Storage config
+    GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
+    gcpOptions.setProject(projectId);
+    gcsUtil = new GcsUtil.GcsUtilFactory().create(gcpOptions);
+
+    // Bigtable config
+    connection = BigtableConfiguration.connect(projectId, instanceId);
+    tableId = "test_" + UUID.randomUUID().toString();
+
+    System.out.println("Setting up integration tests");
+
+    String[] keys = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+    keySplits = new byte[keys.length][];
+    for (int i = 0; i < keys.length; i++) {
+      keySplits[i] = keys[i].getBytes();
+    }
+  }
+
+  private static String getTestProperty(String name) {
+    return checkNotNull(System.getProperty(name), "Required property missing: " + name);
+  }
+
+  @After
+  public void teardown() throws IOException {
+    final List<GcsPath> paths = gcsUtil.expand(GcsPath.fromUri(restoreDir + "/*"));
+
+    if (!paths.isEmpty()) {
+      final List<String> pathStrs = new ArrayList<>();
+
+      for (GcsPath path : paths) {
+        pathStrs.add(path.toString());
+      }
+      this.gcsUtil.remove(pathStrs);
+    }
+
+    connection.close();
+
+    // delete test table
+    BigtableConfiguration.connect(projectId, instanceId)
+        .getAdmin()
+        .deleteTable(TableName.valueOf(tableId));
+  }
+
+  @Test
+  public void testHBaseSnapshotImport() throws Exception {
+
+    try (Connection connection = BigtableConfiguration.connect(projectId, instanceId)) {
```

Comment:
why not use the connection you already setup?

Reply:
You are right, I think this is messed up in the process of copy-pasting... fixed.